src/components/CommitBox.tsx

```diff
@@ -1,114 +1,166 @@
+  commitDescriptionClass,
+  commitButtonClass,
+  commitButtonDisabledClass
+} from '../style/CommitBox';
 
+/**
+ * Interface describing component properties.
+ */
 export interface ICommitBoxProps {
+  /**
+   * Boolean indicating whether files currently exist which have changes to commit.
+   */
   hasFiles: boolean;
-  commitFunc: (message: string) => Promise<void>;
-  settings: ISettingRegistry.ISettings;
+
+  /**
+   * Callback to invoke in order to commit changes.
+   *
+   * @param msg - commit message
+   * @returns a promise which commits changes
+   */
+  onCommit: (msg: string) => Promise<void>;
 }
 
+/**
+ * Interface describing component state.
+ */
 export interface ICommitBoxState {
   /**
-   * Commit message
+   * Commit message summary.
    */
-  value: string;
+  summary: string;
+
+  /**
+   * Commit message description.
+   */
+  description: string;
 }
 
+/**
+ * React component for entering a commit message.
+ */
 export class CommitBox extends React.Component<
   ICommitBoxProps,
   ICommitBoxState
 > {
+  /**
+   * Returns a React component for entering a commit message.
+   *
+   * @param props - component properties
+   * @returns React component
+   */
   constructor(props: ICommitBoxProps) {
     super(props);
     this.state = {
-      value: ''
+      summary: '',
+      description: ''
     };
   }
 
-  /** Prevent enter key triggered 'submit' action during commit message input */
-  onKeyPress(event: any): void {
-    if (event.which === 13) {
-      event.preventDefault();
-      this.setState({ value: this.state.value + '\n' });
-    }
-  }
-
-  /** Initalize commit message input box */
-  initializeInput = (): void => {
-    this.setState({
-      value: ''
-    });
-  };
-
-  /** Handle input inside commit message box */
-  handleChange = (event: any): void => {
-    this.setState({
-      value: event.target.value
-    });
-  };
-
-  /** Update state of commit message input box */
-  commitButtonStyle = (hasStagedFiles: boolean) => {
-    if (hasStagedFiles) {
-      if (this.state.value.length === 0) {
-        return classes(stagedCommitButtonStyle, stagedCommitButtonReadyStyle);
-      } else {
-        return stagedCommitButtonStyle;
-      }
-    } else {
-      return classes(stagedCommitButtonStyle, stagedCommitButtonDisabledStyle);
-    }
-  };
-
+  /**
+   * Renders the component.
+   *
+   * @returns fragment
+   */
   render() {
+    const disabled = !(this.props.hasFiles && this.state.summary);
     return (
-      <form
-        className={stagedCommitStyle}
-        onKeyPress={event => this.onKeyPress(event)}
-      >
+      <form className={commitFormClass}>
+        <input
+          className={commitSummaryClass}
+          type="text"
+          placeholder="Summary (required)"
+          title="Enter a commit message summary"
+          value={this.state.summary}
+          onChange={this._onSummaryChange}
+          onKeyPress={this._onSummaryKeyPress}
+        />
```

Comment:
The advice is a summay of max. 50 characters. This could still be longer that a single line. It may be better to use a `TextareaAutosize` here also (with maybe a `maxRows` attribute).

Reply:
I updated the title attribute to _encourage_ users to enter a single line summary not exceeding 50 characters.

As for using a `textarea`, I ultimately opted against this, following the precedent of GitHub's desktop app which also uses a single line `input` box which horizontally scrolls upon overflow. Personally, I think this may be better in order to "nudge" users to limit their summaries to one line with few(er) characters, while encouraging them to place more extended discussions in the `description` box.

Reply:
Thanks for the detailed answer @kgryte. I think we can merge as is; user feedback will tell if it does not fit.